asllib/doc/ASLSemanticsReference.tex

```diff
@@ -19,160 +283,1634 @@
 
 \chapter{Preamble}
 
-The semantics of ASL are embodied in a function that takes as input an ASL program
-and returns all possible behaviours of that program.
+The semantics of ASL defines all valid behaviors of a given ASL specification.
+More precisely, an ASL specification is first parsed into an \emph{abstract syntax tree},
+or AST, for short. Second, a type checker analyzes the \emph{parsed AST} for correctness and, if successful,
+returns a \emph{static environment} and a \emph{typed AST}. Finally, tools such as interpreters and verifiers
+can operate over the typed AST, based on the definition of the semantics, to test and analyze a given specification.
+
+The abstract syntax, parsed AST, and typed AST are defined in the abstract syntax reference~\cite{ASLAbstractSyntaxReference}.
+The ASL type system, which defines the set of parsed ASTs that are considered valid and how to
+produce the corresponding typed ASTs and static environments is defined in the typing reference~\cite{ASLTypingReference}.
```

Comment:
Those references are not showing up for me, even after compiling several times.

Reply:
How do you compile?
What happens when you do `make ASLTypingReference.pdf`?

Reply:
(note I was talking about ASLSemanticsReference.pdf, not ASLTypingReference.pdf as you mention)

`make asldoc` from herdtools7/ doesn't do the bibliography for ASLSemanticsReference properly.
neither does `make ASLSemanticsReference.pdf` from herdtools7/asllib/doc.

Reply:
Please try it now.